Refactor list item view

gemdev111 opened the pull request:
#39

gemdev111 committed:
Refactored ListItemView, added previews with different states
moved ListItemVIew to Lists folder

gemdev111 committed:
moved ListItemProgressView from ListItemView, since it has more then 1 dependecy.  Also used in TransactionsScene separetly

gemdev111 committed:
Project strucuture update
TextStyle moved from Components package to Style package, added necessary Style package imports to Gem
ListItemView minor change

Done #39

gemdev111 committed:
Merge branch 'main'

gemdev111 committed:
fixed pbroj, unlinked files after vsc

gemcoder21 commented on Packages/Components/Sources/Lists/ListItemView.swift:
```diff
@@ -0,0 +1,391 @@
+    let longTitleExtra = "Long Title Extra Long Title Extra Long Title Extra Long Title Extra"
+    let longSubtitleExtra = "Long Subtitle Extra Long Subtitle Extra Long Subtitle Extra"
+
+    let defaultTextStyle = TextStyle(font: Font.system(.body), color: .primary)
```
maybe add `static let defaultTextStyle` for `ListItemView` ? 

then in the init function you can pass `Self.defaultTextStyle` as default value

gemcoder21 commented on Packages/Components/Sources/Lists/ListItemView.swift:
```diff
@@ -0,0 +1,391 @@
+            )
+        }
+
+        // Long Text States
```
```suggestion
```

gemcoder21 commented on Packages/Components/Sources/Lists/ListItemView.swift:
```diff
@@ -0,0 +1,391 @@
+    let tagTextStyleBlue = TextStyle(font: Font.system(.footnote), color: .blue, background: .blue.opacity(0.2))
+
+    return List {
+        // Basic States
```
```suggestion
```

no need for such comments, it's self explanatory in the section

gemdev111 committed:
Merge branch 'main'

Conflicts:
	Packages/Components/Sources/Lists/ListItemView.swift

gemdev111 committed:
Refactored ListItemView & Text Style added defaults value
ListItemView decorated with TextValueView instead of combo text & stle
Added defaults values for TextStyle, Added TextStyle.Preview - to have ability see colors in packages previews

gemdev111 committed:
added note to TextStyle

gemdev111 committed:
Updated Gem dependency with style init

gemdev111 committed:
Project structure update
refactor->rename TextValueView to TextValue
Crated Types folder, moved Value Types(not views) to Types

gemdev111 committed:
Moved System Image to Styles
Added necessary imports Style, removed SystemImage.infoCircle. Since it's repeated image and we don't need it

gemdev111 committed:
updated initi method with clearer init, added missed values to init

gemdev111 committed:
Merge branch 'main'

Conflicts:
	Gem/Transfer/ViewsModels/ConfirmTransferViewModel.swift

gemdev111 committed:
apr text style updated to correct callout

gemdev111 committed:
Merge branch 'main'

gemdev111 committed:
git submodules updated
